### PR TITLE
Fix type stubs in typing.pyi

### PR DESCRIPTION
**Pull request by DmitriyS**
Signature for `get_type_hints` function looks like `obj, globalns=None, localns=None`

### CODE REVIEW

File: stdlib/2/typing.pyi [hunk 1]
```diff
@@ -385,7 +385,8 @@ class Pattern(Generic[AnyStr]):
 
 # Functions
 
-def get_type_hints(obj: Callable) -> dict[str, Any]: ...
+def get_type_hints(obj: Callable, globalns: Optional[dict],
```
Comment (JelleZijlstra) [inline stdlib/2/typing.pyi]: These should both be `: Optional[Dict[Text, Any]] = ...` (the arguments are optional and the keys must be strings).

**Comment by DmitriyS** [comment]
Ok.
I used `dict[AnyStr, Any]` annotation, bcz `Dict ` leads to an error: `Invalid type "typing.Dict"`

File: stdlib/2/typing.pyi [hunk 2]
```diff
@@ -385,7 +385,8 @@ class Pattern(Generic[AnyStr]):
 
 # Functions
 
-def get_type_hints(obj: Callable) -> dict[str, Any]: ...
+def get_type_hints(obj: Callable, globalns: Optional[dict[AnyStr, Any]],
```
Comment (JelleZijlstra) [inline stdlib/2/typing.pyi]: I don't think AnyStr is right; there's nothing stopping users from using a mix of str and unicode in Python 2, and using bytes in Python 3 almost certainly won't work. That's why I suggested Text before. Also, come to think of it the argument types should probably be Mapping, not just dict.

Comment (DmitriyS) [inline stdlib/2/typing.pyi]: Why should be `{u'key1': 'value1', 'key2': 'value2'}` incorrect for users?
btw python3 with {b'key': 'value'} works.

Comment (JelleZijlstra) [inline stdlib/2/typing.pyi]: To be clear, using AnyStr means that all the keys must be either bytes or Text; you can't have a mix.

Comment (gvanrossum) [inline stdlib/2/typing.pyi]: (The reason is that AnyStr is defined as a type variable; see the [mypy docs](http://mypy.readthedocs.io/en/latest/generics.html#type-variables-with-value-restriction).)

**Comment by gvanrossum** [comment]
There's more to do, see variable `_allowed_types` in typing.py (in particular `obj` can also be a module).

**Pull request merged**